projects/ccf-shared/src/lib/services/globals/tracking.state.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { DataAction, StateRepository } from '@ngxs-labs/data/decorators';
+import { NgxsImmutableDataRepository } from '@ngxs-labs/data/repositories';
+import { State } from '@ngxs/store';
+
+
+export interface TrackingStateModel {
+  allowTelemetry?: boolean;
+}
+
+export const LOCAL_STORAGE_ALLOW_TELEMETRY_KEY = 'ALLOW_TELEMETRY';
+export const INITIAL_TELEMETRY_SETTING  = localStorage.getItem(LOCAL_STORAGE_ALLOW_TELEMETRY_KEY) === null ? undefined
+  : localStorage.getItem(LOCAL_STORAGE_ALLOW_TELEMETRY_KEY)?.toLowerCase() === 'true';
+
+@StateRepository()
+@State<TrackingStateModel>({
+  name: 'tracking',
+  defaults: {
+    allowTelemetry: INITIAL_TELEMETRY_SETTING
+  }
+})
+@Injectable()
+export class TrackingState extends NgxsImmutableDataRepository<TrackingStateModel> {
+
+  @DataAction()
+  setAllowTelemetry(allowTelemetry: boolean): void {
```

Comment:
Wayy too much code duplication here. You should be able to simplify this logic further. Look closer, you should be able to get this down to around 5 lines of code with no duplication.